operator/pkg/controllers/controller.go

```diff
@@ -42,90 +45,53 @@ func (c *GenericController) Reconcile(ctx context.Context, req reconcile.Request
 	// 1. Read Spec
 	resource := c.For()
 	if err := c.Get(ctx, req.NamespacedName, resource); err != nil {
-		if errors.KubeObjNotFound(err) {
+		if errors.IsNotFound(err) {
 			return reconcile.Result{}, nil
 		}
-		return reconcile.Result{}, err
+		return *status.Failed, err
```

Comment:
If these are used as direct objects and not pointers, might be simpler to just make status.Failed a struct where it's defined.